packages/widgetbook_core/lib/src/navigation_tree/widgets/navigation_tree_item.dart

```diff
@@ -0,0 +1,165 @@
+
+class NavigationTreeItem extends StatefulWidget {
+  const NavigationTreeItem({
+    super.key,
+    this.level = 0,
+    this.label,
+    this.content,
+    this.iconData,
+    this.icon,
+    this.onTap,
+    this.isExpanded = false,
+    this.onMoreIconPressed,
+  })  : assert(
+          label != null || content != null,
+          'Either a label or content must be provided',
+        ),
+        assert(
+          icon != null || iconData != null,
+          'Either an icon or iconData must be provided',
+        );
+
+  const NavigationTreeItem.category({
+    super.key,
+    required this.label,
+    this.onTap,
+    this.isExpanded = false,
+    this.onMoreIconPressed,
+  })  : level = 0,
+        iconData = Icons.auto_awesome_mosaic,
+        icon = null,
+        content = null;
+
+  const NavigationTreeItem.package({
+    super.key,
+    required this.label,
+    this.onTap,
+    this.isExpanded = false,
+    this.onMoreIconPressed,
+  })  : level = 0,
+        iconData = Icons.inventory,
+        icon = null,
+        content = null;
+
+  const NavigationTreeItem.folder({
+    super.key,
+    required this.label,
+    this.onTap,
+    this.isExpanded = false,
+    this.level = 0,
+    this.onMoreIconPressed,
+  })  : iconData = Icons.folder,
+        icon = null,
+        content = null;
+
+  const NavigationTreeItem.component({
+    super.key,
+    required this.label,
+    this.level = 0,
+    this.onTap,
+    this.isExpanded = false,
+    this.onMoreIconPressed,
+  })  : iconData = null,
+        icon = const ComponentIcon(),
+        content = null;
+
+  const NavigationTreeItem.useCase({
+    super.key,
+    required this.label,
+    this.level = 0,
+    this.onTap,
+    this.isExpanded,
+    this.onMoreIconPressed,
+  })  : iconData = null,
+        icon = const UseCaseIcon(),
+        content = null;
+
+  static const double indentation = 25;
+  static const double iconSize = 34;
+
+  final int level;
+  final String? label;
+  final Widget? content;
+  final Widget? icon;
+  final IconData? iconData;
+
+  final VoidCallback? onTap;
+  final VoidCallback? onMoreIconPressed;
+  final bool? isExpanded;
+
+  @override
+  State<NavigationTreeItem> createState() => _NavigationTreeItemState();
+}
+
+class _NavigationTreeItemState extends State<NavigationTreeItem> {
+  final _showMoreIconNotifier = ValueNotifier<bool>(false);
+
+  @override
+  Widget build(BuildContext context) {
+    return MouseRegion(
+      onEnter: (_) {
+        _showMoreIconNotifier.value = true;
+      },
+      onExit: (_) {
+        _showMoreIconNotifier.value = false;
+      },
+      child: InkWell(
+        onTap: widget.onTap,
+        borderRadius: BorderRadius.circular(NavigationTreeItem.iconSize),
+        child: Row(
+          children: [
+            Row(
+              mainAxisSize: MainAxisSize.min,
+              children: List.generate(
+                widget.level,
+                (index) =>
```

Comment:
The alignment of the ExpanderIcon seems to be slightly off

![image](https://user-images.githubusercontent.com/16862002/209630603-bbbc8ec7-d08d-4872-af7a-a4e726602b7d.png)

The design does not hit this properly as well but I think the expander Icons should be right below the parents "icon"

Reply:
The red line should "hit" the arrow right in the middle, right now it's hitting the arrow on the rightmost corner

Reply:
Did a minor change to fix this but I'm not so happy with it. The sizing of the Material angle icon is a bit weird and might be causing spacing issues. 
Let me know how it looks now.

Reply:
It still seems to be a bit off especially for the package folder it looks a bit misaligned now

Reply:
<img width="397" alt="image" src="https://user-images.githubusercontent.com/16862002/209637428-95e9a671-2903-4f45-bbbe-114284014e5c.png">

Reply:
I made some changes just to allow for more levels but I think the alignment is good now.